crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,717 @@
+            make::js_parenthesized_expression(
+                make::token(T!['(']),
+                node.clone(),
+                make::token(T![')']),
+            ),
+        )),
+    }
+}
+
+/// If the node is a length method, it returns the object of interest.
+fn get_length_node(node: &AnyJsExpression) -> Option<AnyJsExpression> {
+    let AnyJsExpression::JsStaticMemberExpression(node) = node else {
+        return None;
+    };
+    let member_name = node.member().ok()?;
+    let member_name = member_name
+        .as_js_name()?
+        .value_token()
+        .ok()?
+        .token_text_trimmed();
+    if member_name.text() != "length" {
+        return None;
+    }
+    node.object().ok()
+}
+
+/// AnyJsExpressiion -> Some(i64) if the expression is an integer literal, otherwise None.
+fn get_integer_from_literal(node: &AnyJsExpression) -> Option<i64> {
+    if let AnyJsExpression::JsUnaryExpression(unary) = node {
+        let token = unary.operator_token().ok()?;
+        if token.kind() != T![-] {
+            return None;
+        }
+        return get_integer_from_literal(&unary.argument().ok()?.omit_parentheses())
+            .map(|num| -num);
+    }
+    let AnyJsExpression::AnyJsLiteralExpression(AnyJsLiteralExpression::JsNumberLiteralExpression(
+        number,
+    )) = node
+    else {
+        return None;
+    };
+    let number = number.as_number()?;
+    if number.fract() == 0.0 {
+        Some(i64::try_from(number as i128).ok()?)
+    } else {
+        None
+    }
+}
+
+/// Retrieve the value subtracted from the subtraction expression.
+/// # Examples
+/// ```js
+///    a - b // => Some((a, [b]))
+///    a - b - c // => Some((a, [b, c]))
+/// ```
+fn split_minus_binary_expressions(
+    mut expression: AnyJsExpression,
+) -> Option<(AnyJsExpression, Vec<AnyJsExpression>)> {
+    let mut right_list = vec![];
+
+    while let AnyJsExpression::JsBinaryExpression(binary) = expression {
+        let token = binary.operator_token().ok()?;
+        if token.kind() != T![-] {
+            return Some((AnyJsExpression::JsBinaryExpression(binary), right_list));
+        }
+
+        right_list.push(binary.right().ok()?);
+        expression = binary.left().ok()?;
+    }
+    Some((expression, right_list))
+}
+
+/// Combine the expressions in the list with the addition operator.
+fn make_plus_binary_expression(list: Vec<AnyJsExpression>) -> Option<AnyJsExpression> {
+    list.into_iter().rev().reduce(|left, right| {
+        AnyJsExpression::JsBinaryExpression(make::js_binary_expression(
+            left,
+            make::token(T![+]),
+            right,
+        ))
+    })
+}
+
+/// If the node is a negative index, it returns the negative index.
+/// # Examples
+/// ```js
+///     foo[bar.length - 0] // => None
+///     foo[bar.length - 1] // => Some(-1)
+///     foo[bar.length - 2] // => None
+/// ```
+fn extract_negative_index_expression(
+    member: &AnyJsExpression,
```

Comment:
I checked the call sites. It seems to be possible to directly pass the ownership and thus to avoid the cloning line 428.
```suggestion
    member: AnyJsExpression,
```